music_assistant/providers/phishin/provider.py

```diff
@@ -0,0 +1,1015 @@
+        return self._supported_features
+
+    @property
+    def is_streaming_provider(self) -> bool:
+        """Return True if the provider is a streaming provider."""
+        return True
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = MAX_SEARCH_RESULTS,
+    ) -> SearchResults:
+        """Perform search on Phish.in."""
+        if len(search_query.strip()) < 3:
+            return SearchResults()
+
+        try:
+            endpoint = ENDPOINTS["search"].format(term=search_query)
+            search_data = await api_request(
+                self, endpoint, params={"audio_status": "complete_or_partial"}
+            )
+
+            # If venues are found and we want albums, fetch shows for those venues
+            if MediaType.ALBUM in media_types and search_data.get("venues"):
+                venue_shows: list[dict[str, Any]] = []
+                for venue in search_data.get("venues", []):
+                    venue_slug = venue["slug"]
+                    try:
+                        page = 1
+                        while len(venue_shows) < limit:
+                            shows_data = await api_request(
+                                self, "/shows", params={"venue_slug": venue_slug, "page": page}
+                            )
+
+                            shows_on_page = shows_data.get("shows", [])
+                            if not shows_on_page:
+                                break
+
+                            remaining_slots = limit - len(venue_shows)
+                            venue_shows.extend(shows_on_page[:remaining_slots])
+
+                            current_page = shows_data.get("current_page", 1)
+                            total_pages = shows_data.get("total_pages", 1)
+
+                            if current_page >= total_pages or len(venue_shows) >= limit:
+                                break
+
+                            page += 1
+
+                    except ProviderUnavailableError:
+                        self.logger.warning(
+                            "Failed to fetch shows for venue %s: API unavailable", venue_slug
+                        )
+
+                if venue_shows:
+                    search_data["venue_shows"] = venue_shows
+
+            artists, albums, tracks, playlists = parse_search_results(
+                self, search_data, media_types, search_query
+            )
+
+            return SearchResults(
+                artists=artists[:limit] if MediaType.ARTIST in media_types else [],
+                albums=albums[:limit] if MediaType.ALBUM in media_types else [],
+                tracks=tracks[:limit] if MediaType.TRACK in media_types else [],
+                playlists=playlists[:limit] if MediaType.PLAYLIST in media_types else [],
+            )
+        except (MediaNotFoundError, ProviderUnavailableError):
+            raise
+        except Exception as err:
```

Comment:
```suggestion
        except  (MediaNotFoundError, ProviderUnavailableError, aiohttp.ClientError) as err:
```
gave the aiottp.ClientError as example - just try to catch specific exceptions that can actually occur